rust/agents/scraper/src/scraper/block_cursor.rs

```diff
@@ -0,0 +1,91 @@
+use std::time::{Duration, Instant};
+
+use eyre::Result;
+use sea_orm::prelude::*;
+use sea_orm::{ActiveValue, Insert, Order, QueryOrder, QuerySelect};
+use tokio::sync::RwLock;
+use tracing::{debug, instrument, trace, warn};
+
+use crate::date_time;
+use crate::db::cursor;
+
+const MAX_WRITE_BACK_FREQUENCY: Duration = Duration::from_secs(10);
+
+#[derive(Debug)]
+struct BlockCursorInner {
+    /// Block height
+    height: u64,
+    /// Last time we updated the database with the block height.
+    last_saved_at: Instant,
+}
+
+#[derive(Debug)]
+pub struct BlockCursor {
+    db: DbConn,
+    /// The abacus domain this block cursor is for.
+    domain: u32,
+    inner: RwLock<BlockCursorInner>,
+}
+
+impl BlockCursor {
+    pub async fn new(db: DbConn, domain: u32, default_height: u64) -> Result<Self> {
+        #[derive(Copy, Clone, Debug, EnumIter, DeriveColumn)]
+        enum QueryAs {
+            Height,
+        }
+
+        let height = (cursor::Entity::find())
+            .filter(cursor::Column::Domain.eq(domain))
+            .order_by(cursor::Column::Id, Order::Desc)
```

Comment:
Yeah, technically could be an edge case here. Kinda hastily made this change and technically it should always move forward, but there could be a race condition that makes that not happen if more than one scraper is running. Updating...

One thought though as I do this is ordering by ID is a lot cheaper, but I don't think we really need to worry about the cost that much since it checks so rarely.